src/jit/gentree.cpp

```diff
@@ -17522,6 +17522,11 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad()
     {
         // Some AVX instructions here also have MemoryLoad sematics
 
+        if (HWIntrinsicInfo::isAVX2GatherIntrinsic(gtHWIntrinsicId))
```

Comment:
Why was this inserted here, rather than as part of the rest of the logic flow below?

Reply:
The check would have remained correct and simple by changing the code as such:
```diff
- // Do we have 3 operands?
- if (HWIntrinsicInfo::lookupNumArgs(this) != 3)
+ // Do we have less than 3 operands?
+ if (HWIntrinsicInfo::lookupNumArgs(this) < 3)
```

This also ensures that we aren't adding a "fast-path" for checks that don't need it

Reply:
Good point, will do.